packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -62,7 +63,18 @@ class Datalist extends Component {
 			};
 		}
 
-		const payloadWithSchema = { ...payload, schema: mergedSchema };
+		let payloadWithSchema = {
+			...payload,
+			schema: { ...mergedSchema },
+		};
+
+		if (this.hasTitleMap()) {
+			payloadWithSchema = {
+				...payloadWithSchema,
+				schema: { ...payloadWithSchema.schema, titleMap: this.getTitleMap() },
+			};
+		}
```

Comment:
(Small suggestion that doesn't really change anything, so if you disagree, it's ok)
What should be tested is that if there is a titlemap in state (so from trigger), so we add it.
```suggestion
		const payloadWithSchema = {
			...payload,
			schema: { ...mergedSchema, titleMap: this.state.titleMap || mergedSchema.titleMap },
		};
```